src/components/data_submission/NihAnvilUse.js

```diff
@@ -14,26 +13,14 @@ const nihAnvilUseLabels = {
   no_nhgri_no_anvil: NO_NHGRI_NO_ANVIL,
 };
 
-const allNihAnvilUseFields = [
-  'dbGaPPhsID',
-  'dbGaPStudyRegistrationName',
-  'embargoReleaseDate',
-  'sequencingCenter',
-];
-
 export default function NihAnvilUse(props) {
   const {
     onChange,
-    initialFormData,
+    formData,
     validation,
     onValidationChange,
     updateParentRenderState,
   } = props;
-  const [nihAnvilUse, setNihAnvilUse] = useState(initialFormData?.nihAnvilUse || null);
-
-  const clearFormValues = () => {
```

Comment:
Capturing our discussion from earlier: if the fields aren't reset when selecting options, is it possible to submit data that shouldn't be submitted? For example, if option 1 adds a text box but option 2 hides it and you fill the text box then change to option 2, is the text submitted?

Reply:
This is true across the entire form. I am trying to figure out an elegant way to handle continuity for the user and consistency on hidden fields. I have a theory we can use the JSON schema for this, to remove fields that are not conditionally shown. I will look into AJV (what we use for the frontend) to see if it can do this.